hwy/contrib/sort/vqsort-inl.h

```diff
@@ -1660,11 +1660,21 @@ HWY_NOINLINE void Recurse(D d, Traits st, T* HWY_RESTRICT keys,
   // which implies we anyway skip the right partition due to kWasLast.
   HWY_DASSERT(bound != num || result == PivotResult::kWasLast);
 
-  if (HWY_LIKELY(result != PivotResult::kIsFirst)) {
-    Recurse(d, st, keys, bound, buf, state, remaining_levels - 1);
-  }
-  if (HWY_LIKELY(result != PivotResult::kWasLast)) {
-    Recurse(d, st, keys + bound, num - bound, buf, state, remaining_levels - 1);
+  if (kIsSelect) {
+    if (HWY_LIKELY(result != PivotResult::kIsFirst) && k < bound) {
+      Recurse<true>(d, st, keys, bound, buf, state, remaining_levels - 1, k);
+    } else if (HWY_LIKELY(result != PivotResult::kWasLast) && k >= bound) {
```

Comment:
If k == bound, aren't we done already?

Reply:
Either no, or there's an off-by-one somewhere.

With this fix, the tests pass 1000 runs with random k:
https://github.com/google/highway/compare/136b58efccc870d90b6b1836ad2b29270fac05d8..8915b8caff1e203f9844d05790d37d6354840204#diff-adcb5fe8d4a8b73787547a9bff2dd40a1d60e93147153747d8ed317dfc37d670R1667-R1668

Reply:
Oh, good catch with the k - bound: we're shrinking the interval.

Aha, I think I understand the k==bound case: suppose (10, 20, 40, 30, 31), pivot 30, bound=2. So the 10,20 are skipped, we recurse into 40,30,31 but the first item is not yet correctly placed. So yes, >= is indeed necessary.

Looks good. Are you happy with the testing? How about we do a bit of final cleanup (other comment above, plus adding some documentation on what k means and that it's in [0, num]), then we can merge?